packages/eslint-plugin/tests/rules/prefer-nullish-coalescing.test.ts

```diff
@@ -583,6 +583,60 @@ if (((a = b), b || c)) {
         },
       ],
     },
+    {
+      code: `
+let a: string | undefined;
+let b: string | undefined;
+
+if (!(a || b)) {
+}
+      `,
+      options: [
+        {
+          ignoreConditionalTests: true,
+        },
+      ],
+    },
+    {
+      code: `
+let a: string | undefined;
+let b: string | undefined;
+
+if (!!(a || b)) {
+}
+      `,
+      options: [
+        {
+          ignoreConditionalTests: true,
+        },
+      ],
+    },
+    {
+      code: `
+let a: string | undefined;
+let b: string | undefined;
+
+Boolean(!!(a || b));
+      `,
+      options: [
+        {
+          ignoreBooleanCoercion: true,
+        },
+      ],
+    },
+    {
+      code: `
+let a: string | undefined;
+let b: string | undefined;
+
+Boolean(!(a || b));
```

Comment:
oh, hmm. I hate to do this. But actually, now that I'm thinking harder about this, I don't think we want this. This would be unambiguously better to write as `!Boolean(a || b)`, so the `!` really is the sort of implicit coercion that `ignoreBooleanCoercion` constructor is trying to force you to use `Boolean` on instead.

So sorry to change request changes one way and then request they be undone, but would you mind reverting the `ignoreBooleanCoercion` changes?